packages/shell-api/src/collection.ts

```diff
@@ -2383,34 +2385,69 @@ export default class Collection extends ShellApiWithMongoClass {
     );
   }
 
+  async createSearchIndex(
+    name?: string,
+    definition?: SearchIndexDefinition
+  ): Promise<string>;
+  async createSearchIndex(
+    name?: string,
+    type?: 'search' | 'vectorSearch',
+    definition?: SearchIndexDefinition
+  ): Promise<string>;
+  async createSearchIndex(
+    definition?: SearchIndexDefinition,
+    type?: 'search' | 'vectorSearch'
+  ): Promise<string>;
+  async createSearchIndex(
+    description?: SearchIndexDescription
+  ): Promise<string>;
```

Comment:
I don't think any of the arguments should be optional here, except type in the third one?

Reply:
Then we would need two more overloads:
```
async createSearchIndex(): Promise<string>;
async createSearchIndex(name: string): Promise<string>;
```
Does it make sense to add those? I feel like with the current state it is pretty descriptive.

Reply:
Yes, exactly, those aren't valid overloads so they shouldn't be allowed 👍 As mentioned on Slack, there are four variants we want to/can support:

```
createSearchIndex(definition)
createSearchIndex(name, definition)
createSearchIndex(name, type, definition)
createSearchIndex(description)
```

A definition or description is always necessary.

Reply:
Hmm we have a test tho that claims that just a name is allowed: https://github.com/mongodb-js/mongosh/blob/3456adced8738f135befb3fbc122e1f018bb52e2/packages/shell-api/src/collection.spec.ts#L2584-L2596

Reply:
That test doesn't have anything to do with this function, though

Reply:
Wrong lines. Here it is: https://github.com/mongodb-js/mongosh/blob/3456adced8738f135befb3fbc122e1f018bb52e2/packages/shell-api/src/collection.spec.ts#L2648-L2658

Reply:
Yeah, `definition: {}` is not a valid definition and will be rejected by the server, this isn't a supported use case